TMEDIA-287: Add Overline as an option to the Results List block

nschubach opened the pull request:
## Description
Add Overline as an option to the Results List block

## Jira Ticket
- [TMEDIA-287](https://arcpublishing.atlassian.net/browse/TMEDIA-287)

## Acceptance Criteria
* The Results List block has an additional option to Show overline under Custom Fields > Show promo elements
  * By default, Show overline should be disabled, on both existing Results List blocks on a page, and new Results List blocks placed on the page – and the overline should not render (it should look exactly like it does now)
  * If a PageBuilder user enables the Show overline field for a block, then the Overline should display above the headline (the same way we see it display above other blocks, e.g. Large Promo) 
* Tests and documentation are updated as expected

## Test Steps
1. Checkout this branch `git checkout TMEDIA-287-add-overline-results-list`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/results-list-block`
3. Open a page with the the results list (http://localhost/pagebuilder/editor/curate/?p=pnf2rUzD5HMTW4Bs&v=vQltZF5WeIMTW4Bs)
4. Verify you can enable (disabaled by default) the Overline element from `Show promo elements` config.
5. Verify that changing this true with a valid overline value shows on page.
6. Verify that changing this true with a valid overline and url value shows on page with updated link.

## Effect Of Changes
### Before
![image](https://user-images.githubusercontent.com/2287238/126398324-90cb258f-a045-4f09-9707-7fda3a39de3d.png)
![image](https://user-images.githubusercontent.com/2287238/126398369-de0cfd9f-b656-4ce0-9aee-de245b74d710.png)


### After
![image](https://user-images.githubusercontent.com/2287238/126396668-c8918b3a-3d78-447c-8705-4234ff8d9cf0.png)
![image](https://user-images.githubusercontent.com/2287238/126398434-86da102e-0a56-4daa-b6a4-c38eb08ef86a.png)


## Dependencies or Side Effects
N/A

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

Commit:
TMEDIA-287: Add Overline as an option to the Results List block

matthewroach reviewed (changes requested):
@nschubach - An overline is part of an ANS object (Story), this should be applied to all items in the result list and using the ANS fields.

nschubach commented:
Oh, I see.  I oddly grabbed an example from the manual promo for some reason.  Will rework.

Commit:
Added a per item section overline capability as well as label and sponsor support

nschubach commented:
Added Item overlines, renamed list overline.
![image](https://user-images.githubusercontent.com/2287238/126528283-3897c0df-e75b-4ac6-86bc-0ff3dd543644.png)

nschubach committed:
Merge branch 'canary' into TMEDIA-287-add-overline-results-list

matthewroach commented on blocks/results-list-block/features/results-list/results/index.jsx:
```diff
@@ -185,6 +205,14 @@ const Results = ({
 
   return (viewableElements?.length > 0 && !isServerSideLazy) ? (
     <div className="results-list-container">
+      {showOverline
```
This is not required, as an overline is a story item.

nschubach replied:
removed

matthewroach commented on blocks/results-list-block/features/results-list/results/result-item.jsx:
```diff
@@ -53,9 +55,16 @@ const ResultItem = React.memo(React.forwardRef(({
         )
         : null }
-      {(showHeadline)
+      {(showHeadline || showItemOverline)
```
This logic is causing an issue with toggling the headline, this will pass if either showHeadline or showItemOverline is true and a few lines below the headline is always shown if this evaluates to true

nschubach replied:
fixed

matthewroach reviewed (changes requested):
The overline that's above the result list is not correct. An overline is an ANS item related to stories.
Also, this was not an ask within the ticket? Can we confirm this is required. 
If we are adding what I would refer to as heading to go before the results list we should be using the `<Heading>` component and also be using `<HeadingSection>` to ensure hierarchy is maintained
Also, the custom field options for a list heading does not make sense within the custom field group `Show promo elements`

nschubach commented:
> The overline that's above the result list is not correct. An overline is an ANS item related to stories.
Also, this was not an ask within the ticket? Can we confirm this is required.
If we are adding what I would refer to as heading to go before the results list we should be using the <Heading> component and also be using <HeadingSection> to ensure hierarchy is maintained

Talked to Jenae about it since Sara is out.  She's going to create a ticket if they want that in, but I'm gong to remove it for now given the heading considerations.  She had agreed it was a feature that could probably be used, so I left it in previous.

Commit:
Remove the overall list 'overline' to evaluate it for necessity later

nschubach commented:
![image](https://user-images.githubusercontent.com/2287238/126552472-2be239b0-30f3-44fd-969d-93af52275b1a.png)

matthewroach commented on blocks/results-list-block/features/results-list/default.jsx:
```diff
@@ -25,6 +25,8 @@ const ResultsList = ({ customFields }) => {
   } = useFusionContext();
   const {
+    overline,
+    overlineURL,
```
These will never be present on customFields now it's been removed

nschubach replied:
Yep, I missed removing these

matthewroach commented on blocks/results-list-block/features/results-list/results/helpers.js:
```diff
@@ -26,11 +26,13 @@ const reduceResultList = (state = defaultResultList, action) => {
+    showHeadline: true,
+    showImage: true,
+    showItemOverline: false,
+    showOverline: false,
```
Leftover?

nschubach replied:
and this...

matthewroach commented on blocks/results-list-block/features/results-list/results/helpers.test.js:
```diff
@@ -7,17 +7,35 @@ describe('resolveDefaultPromoElements', () => {
+      showHeadline: true,
+      showImage: true,
+      showItemOverline: false,
+      showOverline: false,
```
Leftover?

nschubach replied:
and this...

Commit:
Forgot to remove some things.

nschubach commented:
Odd that lint didn't pick up on those, but removed those and an unused mock

matthewroach reviewed (approved):
Nice work @nschubach
